### ISSUE TITLE
RedisJSON codec

### ISSUE THREAD

**Issue by adrian-skybaker** [redisson/redisson#4319]
Recently AWS [announced](https://aws.amazon.com/blogs/database/unlocking-json-workloads-with-elasticache-and-memorydb/) native support for JSON in Elasticache/MemoryDB. It's API compatible with the [RedisJSON](https://redis.io/docs/stack/json/) module, which stores JSON in a tree format in a binary structure, and exposes commands to interact with the JSON structures directly, similar to mongodb documents etc.

It would be great if redisson had a codec that supported this format.

**Comment by mrniko** [comment]
Done. https://github.com/redisson/redisson/wiki/6.-Distributed-objects#615-json-object-holder

**Comment by adrian-skybaker** [comment]
This is working well for us - the only oddity I've noticed is that the batch API returns the sync interface (org.redisson.api.RBatch#getJsonBucket). It's not practically an issue since this interface extends the async one, but I wonder if this is a slight error in the API, and it should be returning the async interface only from the batch?

I've added https://github.com/redisson/redisson/issues/4380 to track support for JSON.MGET, which I don't think is covered yet.
